Feature/user model

macnoblebk opened the pull request:
Adds User.cs, creates user model and adds functionality hash and verify password

macnoblebk committed:
Creates User.cs file and defines user model

macnoblebk committed:
Creates User model

macnoblebk committed:
Adds functionality to hash and verify user password

Blyatman0692 committed:
Website menu with dummy login page

nurselaine commented on ContosoCrafts.WebSite/Models/User.cs:
```diff
@@ -0,0 +1,62 @@
+        public DateTime JoinedAt { get; set; }
+        public string Username { get; set; }
+        public string PasswordHash { get; set; }
+        public string ProfilePictureUrl { get; set; }
```
Have we decided how we want to identify user's profile pictures?

macnoblebk replied:
I'm not sure I understand entirely what you are asking but I don't think we have. We can discuss it in the group.

nurselaine replied:
I think we should discuss more about what user data should look like and where the data will go since I think prof mentioned something about databases in class

nurselaine commented on ContosoCrafts.WebSite/Models/User.cs:
```diff
@@ -0,0 +1,62 @@
+            }
+        }
+
+        // Method to verify if the provided password matches the stored hash
```
Where does the password and user data get stored?

macnoblebk replied:
I was thinking it would be written to a file.

Blyatman0692 reviewed (approved):
Let's start with this and work to see if things need to be changed

Blyatman0692 committed:
Merge branch 'T02-Login-Menu' into feature/user-model

Blyatman0692 committed:
added LoadUser and SaveUser to User.cs
added empty users.json to data folder

Blyatman0692 committed:
-Renamed User.cs to BobaUser.cs to avoid naming conflict
-Renamed uers.json to bobausers.json
-Completed register logic
-Completed login logic
